pkg/openapi/map.go

```diff
@@ -10,7 +10,7 @@ func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenA
 	result := generated.GetOpenAPIDefinitions(ref)
 	for _, v := range result {
 		for name := range v.Schema.SchemaProps.Properties {
-			if name == "deployArgs" || name == "buildArgs" || name == "params" {
+			if name == "deployArgs" || name == "buildArgs" || name == "params" || name == "details" {
```

Comment:
Oh, I forgot about this hack.  This explains a lot of errors I see because this list is not comprehensive for all the places we use GenericMap.  There's got to be a better more proper approach for this.

Reply:
I burnt a lot of time trying to sniff out a better way to do this. I eventually stumbled onto https://github.com/kubernetes/kube-openapi/tree/master/pkg/generators#custom-openapi-type-definitions 

... unfortunately, `func (_ GenericMap) OpenAPIDefinition() openapi.OpenAPIDefinition {//...` wasn't doing the trick (curiously enough, implementing the other two methods for `MicroTime` did though).